great_expectations/experimental/datasources/experimental_base_model.py

```diff
@@ -55,12 +80,55 @@ def yaml(
 
         # pydantic json encoder has support for many more types
         # TODO: can we dump json string directly to yaml.dump?
-        intermediate_json = json.loads(self.json())
+        intermediate_json = self._json_dict(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            skip_defaults=skip_defaults,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+            encoder=encoder,
+            models_as_dict=models_as_dict,
+        )
         yaml.dump(intermediate_json, stream=stream_or_path, **yaml_kwargs)
 
         if isinstance(stream_or_path, pathlib.Path):
             return stream_or_path
         return stream_or_path.getvalue()
 
+    def _json_dict(
+        self,
+        *,
```

Comment:
Not sure if adding the `*` to `yaml` would be useful or break anything. Seems like requiring named arguments might be good.

Reply:
I like letting the first argument (`stream_or_path`) be positional.
I agree with making the rest keyword only.

Reply:
Sounds good me! I'm also fine with leaving this as is.